Explorer/Assets/DCL/SDKComponents/NftShape/Frames/Pool/FramesPool.cs

```diff
@@ -0,0 +1,61 @@
+using DCL.ECSComponents;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Pool;
+using Utility;
+
+namespace DCL.SDKComponents.NftShape.Frames.Pool
+{
+    public class FramesPool : IFramesPool
+    {
+        private readonly IReadOnlyDictionary<NftFrameType, AbstractFrame> prefabs;
+        private readonly AbstractFrame defaultPrefab;
+        private readonly Dictionary<NftFrameType, IObjectPool<AbstractFrame>> pools = new ();
+        private readonly Dictionary<AbstractFrame, NftFrameType> types = new ();
+
+        public FramesPool(NftShapeSettings settings) : this(settings.FramePrefabs(), settings.DefaultFrame())
+        {
+        }
+
+        public FramesPool(IReadOnlyDictionary<NftFrameType, AbstractFrame> prefabs, AbstractFrame defaultPrefab)
+        {
+            this.prefabs = prefabs;
+            this.defaultPrefab = defaultPrefab;
+        }
+
+        public AbstractFrame NewFrame(NftFrameType frameType, Transform parent)
+        {
+            var g = Pool(frameType).Get()!;
+            g.transform.SetParent(parent, false);
+            g.transform.SetLocalPositionAndRotation(Vector3.zero, Quaternion.identity);
+            types[g] = frameType;
+            return g;
+        }
+
+        public void Release(AbstractFrame frame)
+        {
+            Pool(types[frame]).Release(frame);
+            types.Remove(frame);
+        }
+
+        private IObjectPool<AbstractFrame> Pool(NftFrameType frameType)
+        {
+            if (pools.TryGetValue(frameType, out var pool) == false)
+            {
+                pool = pools[frameType] = new ObjectPool<AbstractFrame>(
+                    () => Object.Instantiate(Prefab(frameType)),
+                    g => g.gameObject.SetActive(true),
```

Comment:
Do you have a parent you reparent to on `Release`?

Reply:
Good mention, I think it should be reparented to the root